Newsletters: add 251 (2023-05-17)

harding opened the pull request:
- [x] News and PRs @harding 
- [x] Lede, releases/RCs, and topic entries @harding 
- [x] Why do we have a mempool? @glozow 
- [x] PR to be assigned @Xekyo 
- [x] Suggestion: Coredev meeting summary notes (https://github.com/bitcointranscripts/bitcointranscripts/pull/247)

glozow commented on _includes/specials/policy/en/01-why-mempool.md:
```diff
@@ -0,0 +1,74 @@
+feerate, which helps nodes guess which transactions are most likely to
+be confirmed.
+
+Next week's section will dive further into mempool policy and its uses
+in protecting network resources, assisting in safer soft forks,
+avoiding denial of service attacks, and more.
```
(Not to strong-arm what the next section in the series will be)!

glozow reviewed (commented):
Added my "why do we have a mempool" + "why not submit directly to miners"

Xekyo commented on _includes/specials/policy/en/01-why-mempool.md:
```diff
@@ -0,0 +1,74 @@
+stats?).  Validating the transactions is also much faster: the
+signatures and scripts verified (and cached), timelock requirements
+already calculated, and relevant UTXOs loaded from disk if necessary.
+The node can also forward the block onto its other peers very quickly,
```
```suggestion
The node can also forward the block onto its other peers promptly,
```

Xekyo commented on _includes/specials/policy/en/01-why-mempool.md:
```diff
@@ -0,0 +1,74 @@
+a transaction originates from as well as which node may be responsible
+for confirming it. Someone wishing to censor particular entities may
+target miners, popular exchanges, or other centralized submission
+services, but it would be difficult to block anything completely.
```
In this context it might be good to point out that the general availability of unconfirmed transactions also minimizes the entrance cost for anyone to become a miner if they’re dissatisfied with the performance or conduct of the current miners.

Xekyo commented on _includes/specials/policy/en/01-why-mempool.md:
```diff
@@ -0,0 +1,74 @@
+distributed transaction and block relay network. All of these benefits
+are most pronounced when everybody sees all transactions before miners
+include them in blocks - just like any cache, it is most useful when
+it's "hot" and must be limited in size to fit in memory. This leads us
+to our first example of and reason for mempool policy: transaction
+feerate, which helps nodes guess which transactions are most likely to
+be confirmed.
```
How is “transaction feerate” a mempool policy? Are you referring to the dynamic minimum feerate, the `minRelayTxFeerate` or something else?

glozow replied:
I'm referring to the concept of feerate in general

glozow replied:
Ok I agree with you that just the word "feerate" is not a policy. I've reworded the end of the post to be more concise as well. Now that we have a good idea of what the next posts will be, I think this is a good enough transition.

Xekyo reviewed (commented):
Just reviewed Gloria’s mempool piece so far, other review to follow

vostrnad commented on _includes/specials/policy/en/01-why-mempool.md:
```diff
@@ -0,0 +1,74 @@
+allows users to have a better sense of what others are bidding and
+what bids have been successful in the past.
+
+However, there is no such thing as "The Mempool" - each node may have
```
```suggestion
However, there is no such thing as "the mempool"---each node may have
```

vostrnad commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,204 @@
+- **Challenges with zero-conf channels when dual funding:** Bastien
+  Teinturier [posted][] to the Lightning-Dev mailing list about the
+  challenges of allowing [zero-conf channels][] when using the
+  [dual-funding protocol][].   Zero-conf channels can be used even the
```
```suggestion
  [dual-funding protocol][].   Zero-conf channels can be used even before the
```

vostrnad commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,204 @@
+- **Summaries of Bitcoin Core developers in-person meeting:** several
+  developers working on Bitcoin Core recently met to discuss aspects of
+  the project.  Notes from several discussions during that meeting have
+  been [published][].  Topics discussed included [fuzz testing][], an
+  [assumeUTXO][], [ASMap][], [silent payments][], [libbitcoinkernel][],
```
```suggestion
  been [published][].  Topics discussed included [fuzz testing][],
  [assumeutxo][], [ASMap][], [silent payments][], [libbitcoinkernel][],
```

vostrnad commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,204 @@
+      with the current design, provide an overview of the new design,
+      and suggest some of the challenges and tradeoffs involved.  The
+      summary mentions slides or other material which is not available,
+      and there's no written summary that we're available of, so we're
```
```suggestion
      and there's no written summary that we're aware of, so we're
```

vostrnad commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,204 @@
+- [Bitcoin Core 24.1rc2][] is a release candidate for a maintenance
+  release of the current version of Bitcoin Core.
+
+- [Bitcoin Core 25.0rc1][] is a release candidate for the next major
```
There is already a 25.0rc2.

glozow commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,210 @@
+  release of the current version of Bitcoin Core.
+
+- [Bitcoin Core 25.0rc1][] is a release candidate for the next major
+  version of Bitcoin Core.
```
Please also add Bitcoin Core 23.2rc1!

glozow commented on _includes/specials/policy/en/01-why-mempool.md:
```diff
@@ -0,0 +1,74 @@
+target miners, popular exchanges, or other centralized submission
+services, but it would be difficult to block anything completely.
+
+In summary, a mempool is an extremely useful cache that allows nodes
```
> a) receivers get notice of an incoming payment promise

I don't like to cite this as a benefit of mempool due to how frequently it misleads people into thinking there is some degree of finality prior to confirmation.

> b) senders and receivers can interact with these pending payments in order to reprioritize them via RBF and CPFP

Also won't mention this because you don't need a mempool to do these things.

glozow replied:
Eh, I suppose mempool is quite helpful in fee-bumping. Maybe I will formulate this into something like "the general availability of unconfirmed transactions helps create a transparent/public fee market. Users are able to see what other users are bidding in terms of fees, and attach transactions spending unconfirmed outputs to fee-bump via CPFP" ?

glozow replied:
Did a bit of rewording. I think most of this will end up in post number 2^ as discussed.

harding commented on _includes/specials/policy/en/01-why-mempool.md:
```diff
@@ -0,0 +1,79 @@
+corresponding to each transaction, and refuse to accept any
+transactions matching a particular pattern. This type of Bitcoin may
+be more convenient at times, but would be missing a few of Bitcoin's
+most valued properties.
```
I think this is a fine description of one of the benefits of a decentralized relay network, and I don't think it needs to be changed, but I wanted to note that I've always felt that the strongest argument for an independent relay network is to avoid naturally privileging lager miners over smaller miners.  E.g., [this argument](https://gist.github.com/harding/7c6c5215a0aec8e0b8b186889def01d8)

glozow replied:
Really good point, tried to incorporate this a bit.

harding commented on _includes/specials/policy/en/01-why-mempool.md:
```diff
@@ -0,0 +1,78 @@
+cache, a mempool is most useful when it is "hot" and must be limited in size
+to fit in memory.
+Next week's section will explore the use of incentive compatibility as
+a metric for keeping the most useful transactions in their mempools.
```
It's unclear to me who "their" is in this final sentence.  I think the word can be dropped or it can be changed to something more specific (user's / miner's / everyone's).

harding reviewed (commented):
Reviewed why-mempool; looks great to me except for one nit.  Thanks!

Xekyo reviewed (commented):
Juicy newsletter! Rest LGTM, no further comments :+1:

LarryRuane commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,242 @@
+  improving privacy, improving scalability, and reducing fee costs:
+
+    - *Payment forwarding:* rather than Alice paying Bob, Alice instead
+      pays Bob's vendor Carol, reducing a debt he owes her (or
```
```suggestion
      pays Bob's vendor, Carol, reducing a debt he owes her (or
```

LarryRuane commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,242 @@
+
+    - *Batched payment forwarding:* rather than Alice paying Bob, Alice
+      instead pays several people Bob owes money (or wants to establish
+      a credit with).  Gould's example considers an exchange which has a
```
```suggestion
      a credit with).  Gould's example considers an exchange that has a
```

LarryRuane commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,242 @@
+      redesign of how transactions and their metadata are stored in
+      Bitcoin Core's mempool.  The notes describe a number of problems
+      with the current design, provide an overview of the new design,
+      and suggest some of the challenges and tradeoffs involved.  The
+      summary mentions slides or other material which is not available,
+      and there's no written description of which we're aware, so we're
+      unable to provide a more detailed description at this time, but
+      anyone interested in the topic may want to read the summary.
```
This sentence is quite long, here's one way to break it up and simplify it:

```suggestion
      and suggest some of the challenges and tradeoffs involved.  The
      summary mentions slides or other material, but as these are unavailable,
      we're
      unable to provide more detail at this time. But
      anyone interested in the topic may want to read the summary.
```

LarryRuane commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,242 @@
+      anyone interested in the topic may want to read the summary.
+
+    - [Project meta discussion][] summarizes a varied discussion about
+      the projects goals and how to achieve them despite many
```
```suggestion
      the project's goals and how to achieve them despite many
```

LarryRuane commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,242 @@
+  says the schema is case insensitive, testing shows that some platforms
+  are not opening the application assigned to handle `bitcoin:` URIs
+  when an uppercase `BITCOIN:` is passed.  It would be preferable if
+  uppercase was handled correctly, as it allows creating more efficient
```
```suggestion
  uppercase was handled correctly, as it allows the creation of more efficient
```

ClaraShk commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,242 @@
+    interactions to build a reputation, or the reputation data will lag
+    far behind the current state of the network (making it less useful
+    or even allowing reputation to be abused).
+
```
A reply email [0] was sent today. In the email, we write that the number of spenders that use the same route is not the important data point, as nodes only track the reputation of their neighbors. We do recognize that during an attack, a small node will have a difficult time routing through a big node. In times of peace, on the other hand, the small node would not be affected by the new scheme.

[0] https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-May/003958.html

harding replied:
Thanks for the link!  I plan on making a slight change to partly address your point:

```diff
     Once deployed by experimenters, it should become easier to answer
     one of the [constructive criticisms][decker endorsement] of this idea, which is how
-    many payments originate from spenders who frequently use the same
-    peers and roughly the same routes.  If the core users of LN are
+    many forwarded payments would actually receive a boost from the
+    scheme.  If the core users of LN are
     frequently sending payments to each other over many of the same
     routes, and if the reputation system works as planned, then that
     core network will be more likely to keep functioning during a
```

I don't want to over-discuss the protocol here (I'm happy to engage on list or to join one of the calls), but where Decker's comments resonate with me is the idea that there might not be enough repeat engagement to build a significant reputation.  As I understand your protocol, it's deliberately designed to be fragile in order to prevent abuse.  Spender Alice must endorse her own HTLC in the first hop; Bob must be confident enough in Alice to endorse the HTLC in the second hop to Carol; Carol must be confident in Bob on the hop to Dan; etc...  If at any stage, an HTLC isn't endorsed, it will remain unendorsed for the remainder of the hops.

If Alice only sends a few payments a month, what basis will Bob have for endorsing Alice, especially if he risks his relationship with Carol (from whom he might receive many multiples of the forwarding income he receives from Alice's occasional HTLCs)?  Same if Bob only rarely forwards to Carol.

If a very large percentage of LN payments originate from infrequent spenders or cross rarely used forwarding nodes, then it seems possible that the protocol won't work (unless I'm misunderstanding something!).  I don't actually think that's the case---I would guess that there are plenty enough frequent spenders and highly used forwarders for it to work now, and that the mean frequency of spending per node will probably increase over time, but I felt that it's worth highlighting Decker's critique (or at least my perception of it) in advance of experimental data being received.

ClaraShk replied:
Thanks for your comment, and I agree that Decker's critique is worth mentioning. Looking forward to discussing this further in the ml or calls!

ClaraShk commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,242 @@
+  attacks][topic channel jamming attacks].  Most notably, they provided a
+  short [proposed specification][bolts #1071] that could be deployed
+  using an experimental flag, preventing deployments of it from having any effect on
+  interactions with non-participating nodes.
```
Today, an email [0] was added to the thread, listing 4 common misconceptions about the reputation scheme. The email reiterates that:

1.  There is no gossip in the scheme
2. Less active edges will experience a gradually degraded quality of service only during a jamming attack
3. Reputation is difficult to gain, needs to be continuously maintained, and is lost easily
4. Reputation gained during times of low activity cannot be used in times of high activity


[0] https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-May/003958.html

harding commented on _posts/en/newsletters/2023-05-17-newsletter.md:
```diff
@@ -0,0 +1,217 @@
+
+- [Libsecp256k1 #1066][] Abstract out and merge all the magnitude/normalized logic FIXME:harding
+
+- [Libsecp256k1 #1299][] Infinity handling: ecmult_const(infinity) works, and group verification FIXME:harding
```
@sipa thanks!  I think while you were writing your summary above, I pushed a commit which dropped our mention of both PRs.  I'm sorry to have wasted your time.  For background, I usually do a quick skim of all commits added to our monitored projects each week to make a list of anything that might be interesting.  Then I research each of the items on the list until I decide that it's either not interesting or I think I understand it well enough to describe it (or ask one of our other contributors to describe it).

In most weeks, nobody besides me ever sees my list of might-be-interesting PRs; this week was an exception because I was running behind schedule.  Again, thanks for the summary!

harding commented:
Made edits for all feedback, except some minor comments about the mempool section.  Updated releases/RCs and added topic entries.  Thanks everyone for the contributions and feedback!

bitschmidty reviewed (commented):
Great newsletter this week team! One correction

bitschmidty commented:
Corrected a link in the LSP section. Made a small change to the mempool writeup.

Thanks @harding @glozow @Xekyo @sipa @LarryRuane @vostrnad @ClaraShk for your contributions this week!
